samples/nrf9160/mqtt_simple/README.rst

```diff
@@ -0,0 +1,79 @@
+.. _mqtt_simple_sample:
+
+nRF9160: Simple MQTT
+####################
+
+The Simple MQTT sample demonstrates how to easily connect an nRF9160 SiP to an MQTT broker and send and receive data.
+
+Overview
+*********
+
+The sample connects to an MQTT broker and publishes whatever data it receives on the configured subscribe topic to the configured publish topic.
+
+Requirements
+************
+
+* The following development board:
+
+  * nRF9160 DK board (PCA10090)
+
+* :ref:`secure_boot` must be programmed on the board.
+
+  The sample is configured to compile and run as a non-secure application on nRF91's Cortex-M33.
+  Therefore, it requires the :ref:`secure_boot` bootloader that prepares the required peripherals to be available for the application.
+
+Building and running
+********************
+
+This sample can be found under :file:`samples/nrf9160/mqtt_simple` in the |NCS| folder structure.
+
+The sample is built as a non-secure firmware image, thus with ``CONFIG_TRUSTED_EXECUTION_NONSECURE=y`` set in ``prj.conf``.
+It can be programmed independently from the secure boot firmware.
+
+See :ref:`gs_programming` for information about how to build and program the application.
+Note that you must program two different applications as described in the following section.
+
+Programming the sample
+======================
+
+When you connect the nRF9160 development kit to your computer, you must first make sure that the :ref:`secure_boot` sample is programmed:
+
+1. Put the **SW5** switch (marked debug/prog) in the **NRF91** position to program the main controller.
+#. Build the :ref:`secure_boot` sample and program it.
+#. Build the Simple MQTT sample (this sample) and program it.
```

Comment:
@joakimtoe We should mention that this sample needs to be built for `nrf9160_pca10090ns`.
Notice the _ns_ at the end of the board name.

Reply:
Does it need that? What about qemu or other boards with nrf9160?

Reply:
Yes, I'd like to state clearly which board to build this sample for. It makes especially sense for the `nrf9160_pca10090` because it comes in two flavours: secure and non-secure. If it's built for secure (without _ns_ at the end) the sample won't run. If we support more than one board, we could document that in addition.

Reply:
Currently, the requirements section says "  * nRF9160 DK board (PCA10090)".

Reply:
@ru-fu that's correct, I don't think it currently builds for `qemu_x86` or other boards.

Reply:
@lemrey, @ru-fu  this sample is not board specific. It works on all boards that has a nrf9160 and can also be built for qemu. I think its misleading that the DK is specified as a requirement in this case.

Reply:
I have tried to build this sample for `qemu_x86` and it did not build:
"`Assertion failed: GCC_M_CPU must be set to find correct lib`"

Reply:
Yes please, if we are to document this sample as being able to run on more than one board, then we need to provide a working configuration for each supported board.

Reply:
Or we should just document the default board, namely PCA10090, and leave it up to the user to migrate the sample to other boards.